src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -283,12 +299,29 @@ def set_key(cmd,
             raise CLIError(str(exception))
 
         if retrieved_kv is None:
+            if content_type and __is_json_content_type(content_type):
+                try:
+                    # Ensure that provided value is valid JSON. Error out if value is invalid JSON.
+                    value = '""' if value is None else value
+                    value = json.dumps(json.loads(value), separators=(',', ':'))
+                except ValueError:
+                    raise CLIError('Value "{}" cannot be parsed as JSON. Since the content type of this setting is "{}", make sure that the value is valid JSON.'.format(value, content_type))
```

Comment:
>'Value "{}" cannot be parsed as JSON. Since the content type of this setting is "{}", make sure that the value is valid JSON.'.format(value, content_type) [](start = 35, length = 154)

nit: 'Value "{}" is not a valid json object, which mismatches with the content type  "{}" '.format(value, content_type)